testing/regress/hpcc/regression/cleanup.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import os
+import re
+import requests
+import time
+from datetime import datetime
+
+# Configure logger
+logger = logging.getLogger('cleanup') 
+
+# Custom logger formatter for serial numbering of log records
+class SerialNumberFormatter(logging.Formatter):
+    def __init__(self):
+        super().__init__()
+        self.serialNumber = 0
+
+    def format(self, record):
+        self.serialNumber += 1
+        record.msg = str(self.serialNumber) + ". " + record.msg
+        return super().format(record)
+
+# Builds and configures the cleanup logger
+def buildCleanupLogger(logDir, cleanupLogger):
+    cleanupLogger.setLevel(logging.INFO)
+    curTime = time.strftime('%y-%m-%d-%H-%M-%S')
+    logName = "cleanup" + "." + curTime + ".log"
+    logPath = os.path.join(logDir, logName)
+    cleanupHandler = logging.FileHandler(logPath)
+    cleanupHandler.setFormatter(SerialNumberFormatter())
+    cleanupLogger.addHandler(cleanupHandler)
+    return cleanupLogger
+
+# Extracts regress log files corresponding to current run of RTE
+def getRegressLogs(mode, logDirPath, startTime):
+    logFileNamePattern = re.compile(r'^(thor|roxie|hthor|roxie-workunit)\.\d{2}-\d{2}-\d{2}-\d{2}-\d{2}-\d{2}\.log(?!.*exclusion)')
+    with os.scandir(logDirPath) as entries:
+        for entry in entries:
+            if entry.is_file():
+                try:
+                    fileCreationTime = datetime.fromtimestamp(entry.stat().st_mtime)
+                    if fileCreationTime >= startTime:
+                        if logFileNamePattern.match(entry.name):
+                            getWorkunitDetails(entry.path, mode)
+                except OSError as e:
+                    logger.error(f"Error accessing log files {entry.path}: {e}")
+
+
+# Extracts workunit details from log file based on the cleanup mode
+def getWorkunitDetails(logFilePath, mode):
+    with open(logFilePath, 'r') as file:
+        lineList = file.readlines()
+        def extractUrl(line):
+            index = line.find("URL")
+            if index != -1:
+                return line[index + 3:-1]
+            else:
+                return None
+        if mode == "workunits":
+            for line in lineList:
+                url = extractUrl(line) 
+                if url:
+                    deleteWorkunit(url)
+        elif mode == "passed":
+            for i, line in enumerate(lineList):
+                if re.search(r'\bpass\b', line, re.IGNORECASE):
```

Comment:
Why are you using regular expression when this is simple substring check in a string (like in line 53)? It is a bit overkill.

Reply:
We will replace it with a normal substring check.